compiler/rustc_codegen_ssa/src/back/link.rs

```diff
@@ -2582,18 +2549,8 @@ fn add_static_crate<'a>(
         cmd.link_rlib(&fix_windows_verbatim_for_gcc(path));
     };
 
-    // See the comment above in `link_staticlib` and `link_rlib` for why if
```

Comment:
The key question here is whether the LTO case can work correctly if both packed and non-packed native libraries occur in the same rlib.
What happens in that case?
What if the packed library is included with `+verbatim` and its name mimics an object file?

If this can be supported correctly, then the `packed` property is basically orthogonal to the `bundle` property and can be potentially added as a new native lib modifier instead of a global compiler option.
Otherwise all native libs in a rlib need to be converted to packed if one of them is packed.

Reply:
Same question likely applies to `fn link_staticlib` as well.

Reply:
Packed and non-packed native libraries are treated independently, both in one rlib shouldn't cause problems.
If packed library is looking like object file it is skipped. If name collision with some object file occurs - there will be a problem. I can add special suffix for packed libraries. In my opinion better to add meta-file with information about files in rlib archive.